services/data/src/react/components/DataProvider.tsx

```diff
@@ -9,20 +10,41 @@ export interface ProviderInput {
     apiVersion?: number
     children: React.ReactNode
 }
-export const DataProvider = (props: ProviderInput) => {
+
+export const queryClientOptions = {
+    defaultOptions: {
+        queries: {
+            // Disable automatic error retries
+            retry: false,
+            // Retry on mount if query has errored
+            retryOnMount: true,
+            // Refetch on mount if data is stale
+            refetchOnMount: true,
+            // Don't refetch when the window regains focus
+            refetchOnWindowFocus: false,
+            // Don't refetch after connection issues
+            refetchOnReconnect: false,
+        },
+    },
```

Comment:
Would be nice if we allowed apps to customize these (e. g. through settings in the `d2.config.js`?).
I guess it could be a nice feature to set `refetchOnMount` to `false` to prevent unnecessary requests, that way the user could navigate the app and wouldn't have to refetch everytime they change the route.

---

Could be done in the future tho, not a change request

Reply:
Yeah, yeah I think it'll be quite necessary to allow configuration in the future. But for this release I tried to keep it as minimal as possible. Initially I wanted to go for completely non-breaking, but that turned out to be difficult. However, I still tried to stick to as minimal of an external change as possible.